### PR TITLE
Added slice BF16/FP32 FWD/BWD kernels

### PR DESCRIPTION
**Pull request by jakpiase**
### PR types
New features

### PR changes
OPs

### Describe
Added slice BF16/FP32 FWD/BWD kernels.

### CODE REVIEW

**Comment by lidanqing-vv** [comment]
@sfraczek  Hi, if you have time, Could you please help review? Thanks !

**Review by lidanqing-vv — DISMISSED** [review]
LGTM

File: paddle/fluid/operators/slice_op.cc [hunk 1]
```diff
@@ -132,6 +132,29 @@ class SliceOp : public framework::OperatorWithKernel {
       if (platform::is_cuda_pinned_place(in_tensor.place())) {
         return framework::OpKernelType(in_tensor.type(), ctx.device_context());
       }
+
+#ifdef PADDLE_WITH_MKLDNN
+      auto input_data_type =
+          framework::OperatorWithKernel::IndicateVarDataType(ctx, "Input");
+
+      if (this->CanMKLDNNBeUsed(ctx, input_data_type)) {
+        // OneDNN uses blocking format, which cannot be always
```
Comment (wozna) [inline paddle/fluid/operators/slice_op.cc]: You can format to remove so many unnecessary new lines.

File: paddle/fluid/operators/slice_op.cc [hunk 2]
```diff
@@ -278,12 +309,33 @@ class SliceOpGrad : public framework::OperatorWithKernel {
...
-                                       ctx, framework::GradVarName("Out")),
-                                   ctx.device_context());
+    auto input_data_type = framework::OperatorWithKernel::IndicateVarDataType(
+        ctx, framework::GradVarName("Out"));
+
+#ifdef PADDLE_WITH_MKLDNN
+    if (this->CanMKLDNNBeUsed(ctx, input_data_type)) {
+      // OneDNN uses blocking format, which cannot be always
+      // supported with reorders, because if blocked dimension is not divisible
+      // by
```
Comment (wozna) [inline paddle/fluid/operators/slice_op.cc]: Same thing with unnecessary new lines.

File: python/paddle/fluid/tests/unittests/mkldnn/test_slice_mkldnn_op.py [hunk 3]
```diff
@@ -0,0 +1,199 @@
+#   Copyright (c) 2018 PaddlePaddle Authors. All Rights Reserved.
```
Comment (wozna) [inline python/paddle/fluid/tests/unittests/mkldnn/test_slice_mkldnn_op.py]: Please change to 2021

File: python/paddle/fluid/tests/unittests/test_static_save_load_bf16.py [hunk 4]
```diff
@@ -22,6 +22,7 @@
 from paddle.fluid.optimizer import SGDOptimizer
 from paddle.fluid.tests.unittests.test_imperative_base import new_program_scope
 from paddle.fluid.tests.unittests.test_static_save_load import PtbModel
+from paddle.fluid.tests.unittests.op_test import convert_float_to_uint16
```
Comment (wozna) [inline python/paddle/fluid/tests/unittests/test_static_save_load_bf16.py]: `convert_float_to_uint16` is probably not needed anymore.

**Review by wozna — DISMISSED** [review]
I added some comments that are only related to factoring, so feel free to add that later. The rest  LGTM.

File: python/paddle/fluid/tests/unittests/test_static_save_load_bf16.py (hunk 4 above)
Comment (jakpiase) [reply python/paddle/fluid/tests/unittests/test_static_save_load_bf16.py]: You're absolutely right, thank you!

File: python/paddle/fluid/tests/unittests/mkldnn/test_slice_mkldnn_op.py (hunk 3 above)
Comment (jakpiase) [reply python/paddle/fluid/tests/unittests/mkldnn/test_slice_mkldnn_op.py]: Thank you, I'll do that

**Review by wozna — APPROVED** [review]
LGTM

**Review by lidanqing-vv — APPROVED** [review]
LGTM.

**Pull request merged**